### ISSUE TITLE
py3.10 spacing in fits 

### ISSUE THREAD

**Issue by nstarman** [astropy/astropy#12230]
### Description

@saimn 

In python 3.10 the spacing has changed for some fits records.

```
::
    >>> from astropy.io import fits
    >>> filename = fits.util.get_testdata_filepath('ascii.fits')
    >>> hdul = fits.open(filename)
    >>> hdul[1].data[:1]  # doctest: +FLOAT_CMP
Expected:
    FITS_rec([(10.123, 37)],
             dtype=(numpy.record, {'names':['a','b'], 'formats':['S10','S5'], 'offsets':[0,11], 'itemsize':16}))
Got:
    FITS_rec([(10.123, 37)],
             dtype=(numpy.record, {'names': ['a', 'b'], 'formats': ['S10', 'S5'], 'offsets': [0, 11], 'itemsize': 16}))
```

### Expected behavior

Got == Expected or the whitespace is ignored.

### Steps to Reproduce

It's in the CI, so just look at https://github.com/astropy/astropy/pull/12176/checks?check_run_id=3783966378

### System Details

CI, python 3.10 minimal dependencies build.

**Comment by saimn** [comment]
This is due to a change in Numpy: https://github.com/numpy/numpy/pull/19686

**Comment by saimn** [comment]
Fixed by #12231
